packages/contracts-bedrock/src/L2/SoulGasToken.sol

```diff
@@ -0,0 +1,258 @@
+    /// IS_BACKED_BY_NATIVE.
+    function deposit() external payable {
+        require(IS_BACKED_BY_NATIVE, "SGT: deposit should only be called when IS_BACKED_BY_NATIVE");
+
+        _mint(_msgSender(), msg.value);
+    }
+
+    /// @notice batchDepositFor can be called by anyone to deposit native token for SoulGasToken in batch when
+    /// IS_BACKED_BY_NATIVE.
+    function batchDepositFor(address[] calldata _accounts, uint256[] calldata _values) external payable {
+        require(_accounts.length == _values.length, "SGT: invalid arguments");
+
+        require(IS_BACKED_BY_NATIVE, "SGT: batchDepositFor should only be called when IS_BACKED_BY_NATIVE");
+
+        uint256 totalValue = 0;
+        for (uint256 i = 0; i < _accounts.length; i++) {
+            _mint(_accounts[i], _values[i]);
+            totalValue += _values[i];
+        }
+        require(msg.value == totalValue, "SGT: unexpected msg.value");
+    }
+
+    /// @notice batchDepositForAll is similar to batchDepositFor, but the value is the same for all accounts.
+    function batchDepositForAll(address[] calldata _accounts, uint256 _value) external payable {
+        require(IS_BACKED_BY_NATIVE, "SGT: batchDepositForAll should only be called when IS_BACKED_BY_NATIVE");
+
+        for (uint256 i = 0; i < _accounts.length; i++) {
+            _mint(_accounts[i], _value);
+        }
+        require(msg.value == _value * _accounts.length, "SGT: unexpected msg.value");
+    }
+
+    /// @notice withdrawFrom is called by the burner to burn SoulGasToken and return the native token when
+    /// IS_BACKED_BY_NATIVE.
+    function withdrawFrom(address _account, uint256 _value) external {
+        require(IS_BACKED_BY_NATIVE, "SGT: withdrawFrom should only be called when IS_BACKED_BY_NATIVE");
+
+        SoulGasTokenStorage storage $ = _getSoulGasTokenStorage();
+        require($.burners[_msgSender()], "SGT: not the burner");
+
+        _burn(_account, _value);
+        payable(_msgSender()).transfer(_value);
+    }
+
+    /// @notice batchWithdrawFrom is the batch version of withdrawFrom.
+    function batchWithdrawFrom(address[] calldata _accounts, uint256[] calldata _values) external {
+        require(_accounts.length == _values.length, "SGT: invalid arguments");
+
+        require(IS_BACKED_BY_NATIVE, "SGT: batchWithdrawFrom should only be called when IS_BACKED_BY_NATIVE");
+
+        SoulGasTokenStorage storage $ = _getSoulGasTokenStorage();
+        require($.burners[_msgSender()], "SGT: not the burner");
+
+        uint256 totalValue = 0;
+        for (uint256 i = 0; i < _accounts.length; i++) {
+            _burn(_accounts[i], _values[i]);
+            totalValue += _values[i];
+        }
+
+        payable(_msgSender()).transfer(totalValue);
+    }
+
+    /// @notice batchMint is called:
+    ///                        1. by EOA minters to mint SoulGasToken in batch when !IS_BACKED_BY_NATIVE.
+    ///                        2. by DEPOSITOR_ACCOUNT to refund SoulGasToken
```

Comment:
why do we need DEPOSITOR_ACCOUNT since we can add minter as the owner?

Reply:
This was for op-geth to refund SGT, but no longer needed since op-geth updates the slot directly.